### ISSUE TITLE
Local Adjustments should have consistent GUI

### ISSUE THREAD

**Issue by chaimav** [Beep6581/RawTherapee#6234]
I appreciate all of the hard work on the new LA module, but I have a request/comment to implement prior to release. Right now the GUI is not very consistent, particularly with 'scope' and 'strength'. I believe these should show up in the same place for every tool, e.g. the top or bottom.
![scope   strength](https://user-images.githubusercontent.com/47765782/117687726-0e627680-b186-11eb-8d73-a7e4a4e83f8a.JPG)

Also, why is there a frame around some of dehaze? And why is strength inside the frame but not scope?

**Comment by Thanatomanic** [comment]
When working on the changed CSS theme, I've expressed my intention before to make sure to give the GUI a thorough review after the next release. For now, I think we should not put too much attention to this, but your comments are duly noted.

**Comment by Desmis** [comment]
@chaimav  @Thanatomanic @AndyAstbury

This harmonization is not self-evident...

I tried, following Andy's request, to add "strength" sliders to harmonize. But the "strength" slider hides very very different things beyond its name...and in some cases, the slider "scope" is common to several tools...you can't have everything and its opposite :)

For example in the case of "denoise" or "Color and light" strength acts as a percentage of the action, once all actions are built
But for example in the action of "soft-light", or "retinex", it is the algorithm itself - by something other than a percentage that is at stake 

The case shown (Retinex and Dehaze) uses 2 "strength" sliders that have absolutely no relation to each other...except for the name. I just moved the "scope" slider for harmonization purpose.


Jacques

**Comment by AndyAstbury** [comment]
When I first raised the 'strength' subject I was envisaging a SPOT strength control.
Under normal circumstances LA spots are "100%" strength.
When doing LA denoise it can be difficult to get the exact amount of denoise required using scope etc.
But getting it in the ball park with conventional adjustments and then tweaking the strength of the LA spot down a little makes for perfect levels of denoise in tightly controlled areas - at least it works that way for me in Jacques branch.

I originally seized on the idea from the overall strength slider in the Wavelets panel, which I've said before is really useful and something of a "get out of jail free" card if you are a bit heavy handed with the main wavelets slider controls.

Likewise, I find the LA strength sliders a tool for adding a final "finesse" to the spot adjustment.

I do agree about panel positioning and perhaps labeling it SPOT STRENGTH to differentiate it from other adjustments, but I would rather see it IN and badly positioned/labelled than omitted.

Those are my thoughts any way..........

**Comment by chaimav** [comment]
Another option for the spot strength slider name could be 'efficacy'. 
Per [dictionary.com ](https://www.dictionary.com/browse/efficacy) "capacity for producing a desired result or effect; effectiveness:"

Some other options (in approximate descending order of how I would choose them) could be:
[intensity](https://www.dictionary.com/browse/intensity)
[potency](https://www.dictionary.com/browse/potency)
[magnitude](https://www.dictionary.com/browse/magnitude)
[power](https://www.dictionary.com/browse/power)
[capacity](https://www.merriam-webster.com/dictionary/capacity)

**Comment by Desmis** [comment]
First of all I have to thank Andy, for this request, which may at first seem redundant, but in the end is very useful. Among other things, it allows you to apply a combination of tools with different percentages depending on the images and/or the location.

For me, as it is implemented in various places in LA that satisfies me. 
Should the user know that it is a percentage or direct action on the algorithm, I do not think so.

Of course there are other names that reflect a force / percentage action, as pointed out by @chaimav, and in LA (or RT)  you can found also "amount".

I am open to any change - only if it is necessary - it is something easy to do, but before doing it - and I think that there is no fire, it is necessary to think about it. Maybe wait for the release of 5.9 and see what the users think.

jacques

**Comment by chaimav** [comment]
I recommend choosing the best name _before_ 5.9. A stable release should not be subject to having tools renamed. It would be very confusing for users.

**Comment by Desmis** [comment]
OK no problem for me :)

if we list all the tools that you suggest for each of them....You will see that the exercise is difficult - in fact we do semantics (with different languages... what has a dominant meaning in French, is not the same in English...).
Make your proposals

In fact the current labels (and tooltips) have been worked on for over a year now with the precious collaboration of Wayne @waynesutton50 . Because not only did he put the terms in good English (which I am totally incapable of doing), but then he brought the "user" point of view ....with questions often of the "describe to me what it does in practice" type, and only after we choose a label.


But just a detail, I'm away from home from tonight for about a week. I'm not sure if I'll have an internet connection and the time.

But I appreciate this kind of proposals and dialogue.

:)

jacques

**Comment by Thanatomanic** [comment]
I want to reiterate my first remark, because reviewing labels and weighing words is a long process, as Jacques describes
> For now, I think we should not put too much attention to this, but your comments are duly noted.

The default language file needs to frozen at some point so that people can really start providing translations for the upcoming release (see #5664).

**Comment by chaimav** [comment]
I think regardless of the algorithm, if the slider is capable of modulating the overall affect the the spot has, a single term can be used for the slider. As @AndyAstbury said, he just wants add some finesse to the tools
#6221 
I also think it should always be either on the top or bottom of every applicable tool next to scope.

**Comment by waynesutton50** [comment]
The point raised is a good one because in some cases the presence of multiple Strength sliders in the same tool could cause confusion (e.g. in Color & Light). Intensity would be an acceptable alternative but RT already uses both Strength and Amount in the various tools so adding a third term is probably not a good idea. If everybody thinks that we should clarify the function in V5.9, my recommendation would be to use "Overall strength".
Despite all the work that has gone into the labels and tooltips, there is still a lot of room for improvement and there will no doubt be many more suggestions once V5.9 has been released.  If the consensus is not to make any more changes prior to the release of V5.9, then changing "Strength" to "Overall Strength"  in V6.0 should not create any undue problems, especially if the sliders stay in the same position (I see that Jacques has pushed a change for Dehaze & Retinex to harmonize it with the other modules).

**Comment by chaimav** [comment]
I am ok with 'overall strength' but maybe 'tool strength' is better. The word tool is already used in 'add _tool_ to current spot' and 'scope (color _tools_)'

**Comment by waynesutton50** [comment]
The problem as Jacques pointed out is that the slider doesn't necessarily apply to all the functions in the tool so we need to be careful.

**Comment by Desmis** [comment]
I agree with Roel @Thanatomanic ... it is now too late to make label changes...except to delay 5.9 again.

On the other hand, we can take advantage of this "issue" to start a process on the revision of labels / tooltips and documentation (Rawpedia, video, etc.):
Who ? Where ? When ? Why ? How ?, etc.  and it is already not without difficulties.

When I look at the time spent with Wayne @waynesutton50 for LA  and Ciecam, and with Xavier @TechXavAL   for wavelet ... and that we were only two, it leaves dream 

Now, I close my computer..tomorrow I am in Paris with my family...

jacques

**Comment by Beep6581** [comment]
In fact, I will make changes to labels and move or delete tooltips for 5.9 - issue #5664. That issue will cover all tools except for Local Adjustments.

The issue here is that it is not clear which widgets relate to which sub-tool and why widgets are not ordered in a consistent way.

@Desmis why is "Scope" in "Dynamic Range & Exposure" (and in most sub-tools) at the top, but in "Dehaze & Retinex" at the bottom?

There is also an "Inverse" checkbox in some sub-tools. As it affects the whole sub-tool (reduces the number of widgets in that sub-tool when checked), I think it should be placed near the top of the sub-tool, not at the bottom.

Another GUI issue is that to me the toolbox currently in RT looks messy, I don't know where one tool ends and the other begins. But that's not for 5.9 nor for this issue.

**Comment by Desmis** [comment]
I am not at home, but at my children's home with a somewhat problematic installation of my computer.

I’m not sure that I have understood all your comments, however, I have created a branch “lachange” that:
* Changes the tooltip option in RT Preferences to “Off” by default.
* Moves the Inverse checkbox to the top as suggested.
* Changes the global Strength slider (added at Andy Astbury’s request) to “Overall strength”.

In addition "Scope for Dehaze and Retinex" is changed since commit 3cb6e88 (it was an "oversight" during the creation of "Overall strength sliders")

Jacques
